### PR TITLE
I set

### PR DESCRIPTION
**Pull request by vegaro**
This pull request changes the Nodes with information sets. Nodes will only be graphic objects from now on.

### CODE REVIEW

File: html/js/guiutils/Tools.js [hunk 1]
```diff
@@ -13,9 +13,9 @@ GTE.UI = (function (parentModule) {
...
     Tools.prototype.newTree = function() {
-        var root = new GTE.TREE.Node(null);
-        var child1 = new GTE.TREE.Node(root);
-        var child2 = new GTE.TREE.Node(root);
+        var root = new GTE.TREE.ISet(null, 1);
+        var child1 = new GTE.TREE.ISet(root, 1);
+        var child2 = new GTE.TREE.ISet(root, 1);
         GTE.tree = new GTE.TREE.Tree(root);
         GTE.tree.updatePositions();
         // Create a node and draw it
```
Comment (alfongj) [inline html/js/guiutils/Tools.js]: This comment is outdated

File: html/js/tree/ISet.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
+GTE.TREE = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new ISet.
+    * @class
+    * @param {ISet} [parent] Parent ISet. If null, this is root.
+    * @param  {Number} [numOfNodes] Number of nodes of this new information set. If null 1
```
Comment (alfongj) [inline html/js/tree/ISet.js]: The two param names should be without square brackets "[" "]"

File: html/js/tree/ISet.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
+GTE.TREE = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new ISet.
```
Comment (alfongj) [inline html/js/tree/ISet.js]: Maybe write Information Set _(creates a new information set)_ rather than the abbreviation in this comment, for clarity?

File: html/js/tree/ISet.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        return "ISet: " + "parent " + this.parent + "; numOfNodes: " + this.numOfNodes + "; moves: " + this.moves;
+    };
+
+    /**
+    * Function that adds a node to ISet
+    * @param {Node} node Node to add
+    */
+    ISet.prototype.addNode = function (node) {
+        this.nodes.push(node);
+    };
```
Comment (alfongj) [inline html/js/tree/ISet.js]: Are we using this function? If not, I would remove it until we need it.

In any case, this.nodes is null, it hasn't been initilized (we are using by now _this.node_ only)

File: html/js/tree/ISet.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+    /**
+    * Function that finds the move that leads to a child ISet
+    * @param {ISet}   childISet ISet to look for
+    * @return {Number}          The index of the move in the list of
+    *                           moves that contains the given iSet 
+    */
+    ISet.prototype.findMoveThatLeadsTo = function (iSet) {
+        for (var indexInList = 0; indexInList < this.moves.length; indexInList++) {
+            if (this.moves[indexInList].child == iSet) return indexInList;
```
Comment (alfongj) [inline html/js/tree/ISet.js]: Replace == with ===

File: html/js/tree/ISet.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        this.moves.push(new GTE.TREE.Move(this, childISet));
+    };
+
+    /**
+    * Function that finds the move that leads to a child ISet
+    * @param {ISet}   childISet ISet to look for
+    * @return {Number}          The index of the move in the list of
+    *                           moves that contains the given iSet 
+    */
+    ISet.prototype.findMoveThatLeadsTo = function (iSet) {
```
Comment (alfongj) [inline html/js/tree/ISet.js:49]: Take into account that more than one move can lead to the same child ISet, once we introduce ISets with more than one node.

e.g., Here _c_ & _d_ are moves that connect the same parent and children Isets, and same goes for _E_ & _F_
image

A solution that seems promising would be to change this function to findMove_s_ThatLeadTo and return an array of indices. But that wouldn't work either, because more than one ISet can be reached through the same Move from a given parent ISet. e.g. _E_ & _F_ here:

image

In summary, a parent ISet and a move don't identify in a unique way a given ISet. We need to think of a different way of representing the data structure that links all ISets together. I will think more about it tomorrow. Also maybe @stengel has some ideas around this. 

(If I didnt explain it properly, please tell me and we can do a quick skype session)

File: html/js/tree/ISet.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    /**
+    * Function that removes the move that contains a children iSet
+    * @param {ISet} iSetToDelete Child ISet to remove
+    */
+    ISet.prototype.removeChild = function (iSetToDelete) {
+        // Find the move that leads to this ISet
+        var indexInList = this.findMoveThatLeadsTo(iSetToDelete);
+        if (indexInList > -1) {
+            this.moves.splice(indexInList, 1);
+        }
```
Comment (alfongj) [inline html/js/tree/ISet.js:65]: This will need to be changed once we figure out what to do with the previous comment.

File: html/js/tree/Move.js [hunk 8]
```diff
@@ -20,9 +20,12 @@ GTE.TREE = (function (parentModule) {
         return "Move: " + "parent: " + this.parent + "; child: " + this.child;
     };
 
+    /**
+    * Draws the move (a line) in the canvas
```
Comment (alfongj) [inline html/js/tree/Move.js]: Please leave a TODO comment here reminding that this function will need to be rewritten once we have >1 nodes per iset.

Maybe you can reuse an existing Issue (whatever you prefer)

File: html/js/tree/ISet.js (hunk 2 above)
Comment (alfongj) [inline html/js/tree/ISet.js]: Or, is this maybe a standard with constructor parameters? (I don;t remember seeing it before) If so leave them :)

File: html/js/tree/Node.js [hunk 9]
```diff
@@ -4,125 +4,36 @@ GTE.TREE = (function (parentModule) {
     /**
     * Creates a new Node.
     * @class
-    * @param {Node} [parent] Parent node. If null, this is root.
+    * @param {ISet} [iSet] iSet that contains the node. Needed for sending onClicks to it
+    * @param {Number} [level] Level of the node
```
Comment (alfongj) [inline html/js/tree/Node.js]: If in the ISet.js constructor you remove the square brackets from parameter names, do it here too, for consistency.

File: html/js/tree/Node.js [hunk 10]
```diff
@@ -4,125 +4,36 @@ GTE.TREE = (function (parentModule) {
     /**
     * Creates a new Node.
```
Comment (alfongj) [inline html/js/tree/Node.js]: Please give a little description on what a Node instance represents.

File: html/js/tree/Tree.js [hunk 11]
```diff
@@ -12,34 +12,45 @@ GTE.TREE = (function (parentModule) {
...
+    };
+
+    /**
+    * Recursive function that draws the Game in the global canvas
+    * starting from a given iSet. If no param is given it will
+    * start from root.
     * Stopping criteria: that the current node is a leaf
     * Recursive expansion: to all of the node's children
-    * @param {Node} [node] Node to start drawing from
+    * @param {ISet} [iSet] ISet to start drawing from
```
Comment (alfongj) [inline html/js/tree/Tree.js:40]: Same thign about square brackets

File: html/js/tree/Tree.js [hunk 12]
```diff
@@ -122,52 +136,57 @@ GTE.TREE = (function (parentModule) {
...
+     * @param {ISet} iSet iSet whose children will be deleted
      */
-    Tree.prototype.deleteChildrenOf = function (node) {
+    Tree.prototype.deleteChildrenOf = function (iSet) {
         // Delete everything below every child
-        while(node.children.length !== 0){
-            this.recursiveDeleteChildren(node.children[0]);
+        while(iSet.moves.length !== 0){
+            // Send moves[0] because moves[] will decrease in size
+            this.recursiveDeleteChildren(iSet.moves[0].child);
```
Comment (alfongj) [inline html/js/tree/Tree.js]: I would do instead:

```
this.recursiveDeleteChildren(iSet.moves.pop().child);
```

**Comment by alfongj** [comment]
Please take a look at the comments. I am going to go to sleep now but I may give it another look tomorrow. 

The most relevant one is the one with the images. I think that once we address that properly we may have to change more things than the ones I mentioned in the comments so far.

File: html/js/tree/ISet.js (hunk 2 above)
Comment (vegaro) [inline html/js/tree/ISet.js]: The brackets are used when the params are optional. In this case, it is true parent is not optional, but numOfNodes is :+1:

File: html/js/tree/ISet.js (hunk 4 above)
Comment (vegaro) [inline html/js/tree/ISet.js]: I will remove it. I wrote it and then thought we didn't need it. It is so simple I can write again in the future.

File: html/js/tree/Tree.js (hunk 11 above)
Comment (vegaro) [inline html/js/tree/Tree.js:40]: Same thing. In this case, if iSet is undefined it will be set to the root, so we can say it is an optional parameter, therefore the brackets make sense.
